pkgs/development/compilers/ghc/common-hadrian.nix

```diff
@@ -367,14 +384,17 @@ let
 
   # TODO(@sternenseemann): is buildTarget LLVM unnecessary?
   # GHC doesn't seem to have {LLC,OPT}_HOST
-  toolsForTarget = [
-    (
-      if targetPlatform.isGhcjs then
-        pkgsBuildTarget.emscripten
-      else
-        pkgsBuildTarget.targetPackages.stdenv.cc
-    )
-  ] ++ lib.optional useLLVM buildTargetLlvmPackages.llvm;
+  toolsForTarget =
+    [
+      (
+        if targetPlatform.isGhcjs then
+          pkgsBuildTarget.emscripten
+        else
+          pkgsBuildTarget.targetPackages.stdenv.cc
+      )
+    ]
+    ++ lib.optional useLLVM buildTargetLlvmPackages.llvm
+    ++ lib.optional useLdLld buildTargetLlvmPackages.bintools;
```

Comment:
This is wrong. `buildTargetLlvmPackages` is the GHC specific llvm package set for `lld`/`opt`, mostly. For `lld`, we need to use the global default `llvmPackages` set so we are using the lld version chosen for the platform. We also should use lld from stdenv if the platform uses it by default.